From fd690a388c77e5d4695914a6ace13ac2ca282f10 Mon Sep 17 00:00:00 2001
From: G.h. Gao <guanhua.gao@nxp.com>
Date: Mon, 10 Sep 2018 13:53:17 +0800
Subject: [PATCH 434/448] Fix compiling error

Fix compiling error introduced by IMA-EVM feature.
Replace dereference_key_locked with rcu_dereference_key in linux-4.9.

Integrated-by: Guanhua Gao <guanhua.gao@nxp.com>
Signed-off-by: Udit Agarwal <udit.agarwal@nxp.com>
---
 security/keys/secure_key.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/security/keys/secure_key.c b/security/keys/secure_key.c
index a6bb20a..6f7a78f 100644
--- a/security/keys/secure_key.c
+++ b/security/keys/secure_key.c
@@ -242,7 +242,7 @@ static long secure_read(const struct key *key, char __user *buffer,
 	char *bufp;
 	int i;
 
-	p = dereference_key_locked(key);
+	p = rcu_dereference_key(key);
 	if (!p)
 		return -EINVAL;
 
-- 
1.7.1

